lua/core/mappings.lua

```diff
@@ -152,56 +155,59 @@ end
 M.bufferline = function()
    local m = plugin_maps.bufferline
 
-   map("n", m.next_buffer, ":BufferLineCycleNext <CR>")
-   map("n", m.prev_buffer, ":BufferLineCyclePrev <CR>")
+   map("n", m.next_buffer, "<cmd>:BufferLineCycleNext <CR>")
+   map("n", m.prev_buffer, "<cmd>:BufferLineCyclePrev <CR>")
 end
 
 M.comment = function()
    local m = plugin_maps.comment.toggle
-   map("n", m, ":lua require('Comment.api').toggle_current_linewise()<CR>")
-   map("v", m, ":lua require('Comment.api').toggle_linewise_op(vim.fn.visualmode())<CR>")
+   map("n", m, "<cmd>:lua require('Comment.api').toggle_current_linewise()<CR>")
```

Comment:
shouldn't this be moved to function syntax ? rather than :lua ....

Reply:
I'm not sure how to get the range stuff working with lua commands. Putting it in the lua format broke the commenting. I can check if there's a easy way though.

Reply:
There does not seem to be one. This plugin's source code is an absolute mess of vimscript. I could implement a better one in lua if one doesn't exist. For now, leave it as is. I'll commit with a fix if I find one. That, or I'll find a better plugin.

Reply:
Wait i gotta commit amend this, group comment toggle is broken... again...

https://github.com/b3nj5m1n/kommentary/tree/main/lua looks like a good choice.

Reply:
Just remove the <cmd> from front, it will work but will just print the commands too, it wasn't working before or something ?

Reply:
So the way I had it was exactly as it was before. Apparently it behaves differently with vim.keymap. This absolutely gross mapping is what's in the plugin's docs and that string finally seems to work so this is what we will do for now.

vim.keymap.set('x', '<C-_>', '<ESC><CMD>lua require("Comment.api").toggle_linewise_op(vim.fn.visualmode())<CR>')

Even the other plugin I linked seems to have similar issues. These people know that you can get marks from lua right? They don't have to get ranges through the vim api, but all of them do.... I checked 3 different plugins. Absolutely ridiculous. I may have to write the 10 millionth comment plugin to finally have a sane api.

Reply:
@zbirenbaum switch to a better commenting plugin instead

Reply:
@siduck see discussion above. I checked around. There are better(ish) ones but they actually don't seem to preserve our current commenting behavior. I think this should be merged and I can look into it later.

Judging by the api's the 3 I checked out offer, I may have to write it myself.

Reply:
@zbirenbaum what about this? https://github.com/terrortylor/nvim-comment